### ISSUE TITLE
Outline of work from slide

### ISSUE THREAD

**Issue by jbkalmbach** [LSSTDESC/bandaid#1]
![image](https://user-images.githubusercontent.com/4755090/92290853-97743000-eeca-11ea-91bb-e1669ef1ceb8.png)

I was thinking about an outline of the work we need to do and I think it naturally breaks up into 3 different modules.

1. Template building module
2. Matching Module
3. Color Recalculation module

For **1** I think the concrete steps with the necessary pieces of code are:

* Build a set of BC03 templates across a grid of Z, tau and *t* values.
  * Code: Script to run BC03 and generate templates
* Develop the search grid
  * Code: Function to redshift SEDs and calculate colors in each redshift shell
  * Code: Data structure to store the properties we want in the matching for each template in each redshift shell
* Build KDE for searching
  * Code: Function to run KDE builder for each redshift shell

That's what I have for now. I will start working on this first module in a branch. Comments appreciated!

**Comment by evevkovacs** [comment]
@sschmidt23 @jbkalmbach  I ran a test using the production pipeline for the skysim5000 catalog (almost the same as cosmoDC2) and tried 2 options for matching the Galacticus galaxies: 1) always choose the best match in (M_r, g-r, r-i ) space 2) always choose the worst match (ie the 10th NN). The idea was to see if the tracks generated in color-z space looked different. Here are some plots showing that the tracks look identical (in slope and width). This being the case, I suspect any attempt to mitigate the discreteness effects by taking linear combinations of Galacticus matches would not help much. 
![image](https://user-images.githubusercontent.com/7797837/92658444-37133300-f2bc-11ea-8b4c-e7871db06c11.png)
![image](https://user-images.githubusercontent.com/7797837/92658534-6164f080-f2bc-11ea-9a1c-3c834d55063a.png)

**Comment by evevkovacs** [comment]
The above plots look absolutely identical. I went back to the runs to make sure that I didn't inadvertently run the same thing twice. If I look at some of the detailed distributions produced by the run (for example, the distributions of rest-frame g-r colors  in slices of M_r for the Galacticus matches, I see very small differences for the brightest M_r band, but no difference for the fainter bands, consistent with the above plots.

**Comment by sschmidt23** [comment]
Thanks, Eve.  Hmm, so I guess that confirms that even 10 nearest neighbors from the SED set have essentially the same colors.  Can you tell what causes this? By causes, I mean why do objects that are reported as distinct have the same color?  Is it very slight differences in some parameter like age or SFR, but the values are actually very similar, or are there just a very small number of templates at high redshift, less than 10 in the neighborhood of all galaxies?  I have the link to the notebook that you sent, I will try to poke around a bit more to answer that question.

**Comment by evevkovacs** [comment]
Yes, I'm not sure. I believe that Dan made some plots of the kde tree search  distance distributions. I'll take a look in those. They may be informative.
